pkg/deploy/lattice/listener_manager.go

```diff
@@ -154,52 +157,42 @@ func (d *defaultListenerManager) List(ctx context.Context, serviceID string) ([]
 			Name:     r.Name,
 		}
 		sdkListeners = append(sdkListeners, &listener)
-
 	}
 
 	return sdkListeners, nil
-
 }
 
-func (d *defaultListenerManager) findListenerByNamePort(ctx context.Context, serviceID string, port int64) (*vpclattice.ListenerSummary, error) {
-	glog.V(6).Infof("calling findListenerByNamePort serviceID %v port %d \n", serviceID, port)
+func (d *defaultListenerManager) findListenerByNamePort(
+	ctx context.Context,
+	serviceId string,
+	port int64,
+) (*vpclattice.ListenerSummary, error) {
 	listenerListInput := vpclattice.ListListenersInput{
-		ServiceIdentifier: aws.String(serviceID),
+		ServiceIdentifier: aws.String(serviceId),
 	}
 
 	resp, err := d.cloud.Lattice().ListListenersWithContext(ctx, &listenerListInput)
+	if err != nil {
+		return nil, err
+	}
 
-	if err == nil {
-		for _, r := range resp.Items {
-			glog.V(6).Infof("findListenerByNamePort>> output port %v item: %v \n", port, r)
-			if aws.Int64Value(r.Port) == port {
-				glog.V(6).Infof("Listener %s Port %v already exists arn: %v \n", serviceID, port, r.Arn)
-				return r, nil
-
-			}
-
+	for _, r := range resp.Items {
+		if aws.Int64Value(r.Port) == port {
+			d.log.Debugf("Port %d already in use by listener %s for service %s", port, *r.Arn, serviceId)
+			return r, nil
 		}
-	} else {
-		return nil, err
 	}
 
-	return nil, errors.New("Listener does not exist")
+	return nil, errors.New("listener does not exist")
```

Comment:
good to have listener details in error message, unless caller of this function wrap error with these details

Reply:
Fixed next PR!